File: cpp_ws/ProgettoCpp/src/Negozio.cpp
#include "Negozio.h"

namespace attivita {

Negozio::Negozio(string m, Time oA, Time oC, int v, int a, TipoNegozio tN)
		:Attivita::Attivita(m, oA, oC, v, a), tipo(tN){}

Negozio::Negozio(string m, Time oA, Time oC, TipoNegozio tN)
		:Negozio::Negozio(m, oA, oC, 0, 0, tN){}

string Negozio::toString(){
	return "Negozio di "  + enumToString[static_cast<int>(tipo)] + ": " + Attivita::toString();
}

Negozio::~Negozio() {}

} /* namespace attivita */

File: ProgettoJava/src/attivita/Negozio.java
package attivita;

import java.time.LocalTime;

public class Negozio extends Attivita{
	final TipoNegozio tipo;
	
	//Costruttori
	public Negozio(String nome, LocalTime orarioApertura, LocalTime orarioChiusura,
			int valutazione, int affollamento, TipoNegozio tipo)
			throws ValutazioneOutOfRangeException, AffolamentoOutOfRangeException {
		super(nome, orarioApertura, orarioChiusura, valutazione, affollamento);
		this.tipo = tipo;
	}
	
	public Negozio(String nome, LocalTime orarioApertura, LocalTime orarioChiusura, TipoNegozio tipo)
			throws ValutazioneOutOfRangeException, AffolamentoOutOfRangeException {
		this(nome, orarioApertura, orarioChiusura, 0, 0, tipo);
	}
	
	//Metodi
	@Override
	public String toString() {
		return "Negozio " + tipo + ": " + super.toString();
	}
	
	public TipoNegozio getTipo() {
		return tipo;
	}
}
